### PR TITLE
Implemented sync status publisher/ updated get status consumer

### PR DESCRIPTION
**Pull request by EmmanuelAv360**
- Implemented a synchronization status publisher
- Sync processes are not set aborted by getting status call anymore

### CODE REVIEW

File: src/UserProfileService.Sync/Services/SynchronizationService.cs [hunk 1]
```diff
@@ -45,23 +48,27 @@ public class SynchronizationService : ISynchronizationService
     /// <param name="scheduleService">Service to manage sync schedule.</param>
     /// <param name="synchronizer"> The service used for process synchronization</param>
     /// <param name="sagaRepositoryContextFactory">Factory to create context to access <see cref="ProcessState" /> data.</param>
+    /// <param name="syncConfiguration">Factory to create context to access <see cref="ProcessState" /> data.</param>
```
Comment (felixkmh) [inline src/UserProfileService.Sync/Services/SynchronizationService.cs:51]: Copied parameter description

File: src/UserProfileService.Sync/Services/SynchronizationService.cs [hunk 2]
```diff
@@ -202,13 +209,21 @@ public async Task<ProcessDetail> GetDetailedProcessAsync(Guid id, CancellationTo
...
-            _logger.LogInfoMessage("Sync lock object is released, sync status will be send ", LogHelpers.Arguments());
+            _logger.LogInfoMessage("Sync lock object is released, sync status will be returned ", LogHelpers.Arguments());
 
-            foreach (ProcessView processView in processViews)
+            if (processViews.Any())
             {
-                await DeclareProcessAbortedAsync(processView.CorrelationId, cancellationToken);
+                bool areProcessesHanging = processViews.Any(
+                    p => p.LastActivity != null
+                         && DateTime.Compare(DateTime.UtcNow, p.LastActivity.Value.AddMinutes(_syncConfiguration.DelayBeforeTimeoutForStep)) > 0);
```
Comment (felixkmh) [inline src/UserProfileService.Sync/Services/SynchronizationService.cs:220]: Is there another value to check if `LastActivity` is `null`? Like a creation time. Or are we safe to assume that `LastActivity` will at least be set once when initialized?

File: src/UserProfileService.Sync/Services/SyncStatusPublisher.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    /// <summary>
+    /// Asynchronously executes the synchronization process in the background.
+    /// </summary>
+    /// <param name="stoppingToken">A <see cref="CancellationToken"/> to observe for cancellation requests.</param>
+    /// <returns>A task representing the asynchronous operation.</returns>
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        _logger.EnterMethod();
+
+        using IServiceScope scope = _serviceProvider.CreateScope();
```
Comment (stw914) [inline src/UserProfileService.Sync/Services/SyncStatusPublisher.cs:56]: I would suggest to get fresh instances of scoped services on every iteration instead to hold them till end of application lifetime.

File: src/UserProfileService.Sync/Services/SynchronizationService.cs (hunk 2 above)
Comment (EmmanuelAv360) [reply src/UserProfileService.Sync/Services/SynchronizationService.cs:220]: LastActivity is set by process initialization

**Review by felixkmh — APPROVED** [review]

**Review by stw914 — APPROVED** [review]

**Pull request merged**